src/analysis/retail/priest/holy/modules/talents/BottomRow/SearingLight.tsx

```diff
@@ -0,0 +1,73 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent, RemoveDebuffEvent, ApplyDebuffEvent } from 'parser/core/Events';
+import TALENTS from 'common/TALENTS/priest';
+import SPELLS from 'common/SPELLS';
+import { calculateEffectiveDamage } from 'parser/core/EventCalculateLib';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+
+const DAMAGE_INCREASE_PER_RANK = [0, 0.25, 0.5];
+
+//Example log: /report/VXr2kgALF3Rj6Q4M/11-Mythic+Anduin+Wrynn+-+Kill+(5:12)/Litena/standard/statistics
+class SearingLight extends Analyzer {
+  damageBonus = 0;
+  damageFromTalent = 0;
+
+  targetsWithHolyFire: Array<number> = [];
+
+  constructor(options: Options) {
+    super(options);
+    if (!this.selectedCombatant.hasTalent(TALENTS.SEARING_LIGHT_TALENT)) {
+      this.active = false;
+    }
+    this.damageBonus =
+      DAMAGE_INCREASE_PER_RANK[this.selectedCombatant.getTalentRank(TALENTS.SEARING_LIGHT_TALENT)];
+
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.SMITE), this.onSmite);
+
+    this.addEventListener(
+      Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.HOLY_FIRE),
+      this.onHolyFireDebuff,
+    );
+
+    this.addEventListener(
+      Events.removedebuff.by(SELECTED_PLAYER).spell(SPELLS.HOLY_FIRE),
+      this.onHolyFireFallOff,
+    );
+  }
+
+  onSmite(event: DamageEvent) {
+    this.damageFromTalent += calculateEffectiveDamage(event, this.damageBonus);
+    return;
+  }
+
+  onHolyFireDebuff(event: ApplyDebuffEvent) {
+    this.targetsWithHolyFire.push(event.targetID);
+  }
+  onHolyFireFallOff(event: RemoveDebuffEvent) {
+    const index = this.targetsWithHolyFire.indexOf(event.targetID);
+    //Every element of the array should be unique so we don't need to delete multiple times
+    //The target should also always be in the array but still error checking
+    if (index > -1) {
+      this.targetsWithHolyFire.splice(index, 1);
+    }
+  }
```

Comment:
A few problems here, and a few ways to fix it.

Problem 1: `event.targetID` isn't unique for enemies, you also have to factor in `event.targetInstance` (different mobs with the same name will have the same targetID but different targetInstance). There is a helper method to make this easier - make the type a string and use `encodeEventTargetString(event)`.

Problem 2: A `Set` is much better suited for this than an array. You can just use the `add` method when a debuff is applied and the `delete` method when it falls, and can check for presence with `has`. For an example of this, see Feral's `SuddenAmbush` module.

Problem 3: While the Set handling works, you actually could drop this way entirely and use the `Enemies` module to check for buff presence. (I'm ok if you want to keep the manual tracking here, just want you to be aware of the other way). You'd include `Enemies` in the module dependencies
```ts
static dependencies = {
    enemies: Enemies,
  };
  protected enemies!: Enemies;
```
Then `onSmite` you can check for buff presence:
```ts
const target = this.enemies.getEntity(event);
if (target !== null && target.hasBuff(SPELLS.HOLY_FIRE)) {
   ... damage calculate
```

Reply:
That is definitely a lot easier, ended up just using the enemies module.